dvc/ignore.py

```diff
@@ -0,0 +1,81 @@
+from fnmatch import fnmatch
+import os
+
+
+class DvcIgnore(object):
+    def __call__(self, *args, **kwargs):
+        assert len(args) == 3
+        root, dirs, files = args
+
+        return self._filter(root, dirs, files)
+
+    def _filter(self, root, dirs, files):
+        raise NotImplementedError
+
+
+class DvcIgnoreFromFile(DvcIgnore):
+    def __init__(self, ignore_file_path):
+        self.dirname = os.path.normpath(os.path.dirname(ignore_file_path))
+        self.patterns = self._read_lines_ignore_comments(ignore_file_path)
+
+    def _filter(self, root, dirs, files):
+
+        files = [f for f in files if not self.matches(root, f)]
+        dirs = [d for d in dirs if not self.matches(root, d)]
+
+        return dirs, files
+
+    def matches(self, root, basename):
+        full_path = os.path.join(root, basename)
+        rel_path = os.path.relpath(full_path, self.dirname)
+
+        for pattern in self.patterns:
```

Comment:
Patterns that we are using here should be unixpaths, right? But you are using os.path above, so on windows it is going to be ntpath. In your tests, you are only testing on unixpaths, which is probably why it works on windows. Or am I missing something?

Reply:
EDIT: you are indeed using os.path in tests, sorry didn't notice right away 🙂Is it fnmatch that handles path transformations?

Reply:
Ineed, fnmatch handles patterns:
https://github.com/python/cpython/blob/cb0748d3939c31168ab5d3b80e3677494497d5e3/Lib/fnmatch.py#L22

Reply:
@pared Got it! Thanks!